File: DarkOrbit 7.5.3/Game/Objects/Activatables/BattleStation.cs
using Ow.Game.Clans;
using Ow.Game.Movements;
using Ow.Net.netty.commands;
using Ow.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ow.Game.Objects.Activatables
{
    class BattleStation : Activatable
    {
        private static String ASSET_NAME = "Clan Battle Station";
        private static short ASSET_TYPE_CLAN_OWNING = AssetTypeModule.BATTLESTATION;
        private static short ASSET_TYPE_NO_CLAN = AssetTypeModule.ASTEROID;
        private static short DESIGN_ID = 2;

        public List<StationModuleModule> inventoryStationModule = new List<StationModuleModule>();
        public List<StationModuleModule> equippedStationModule = new List<StationModuleModule>();

        public BattleStation(Spacemap spacemap, int factionId, Position position, Clan clan) : base(spacemap, factionId, position, clan)
        {
            InitiateModules();
            PrepareSatellites();
        }

        public void InitiateModules()
        {
            equippedStationModule.Add(new StationModuleModule(Id, 2, 1, StationModuleModule.LASER_HIGH_RANGE, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 2, StationModuleModule.LASER_HIGH_RANGE, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 3, StationModuleModule.LASER_HIGH_RANGE, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 4, StationModuleModule.LASER_HIGH_RANGE, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 5, StationModuleModule.DAMAGE_BOOSTER, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 6, StationModuleModule.DAMAGE_BOOSTER, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 7, StationModuleModule.DAMAGE_BOOSTER, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
            equippedStationModule.Add(new StationModuleModule(Id, 2, 8, StationModuleModule.DAMAGE_BOOSTER, 50000, 50000, 50000, 50000, 1, "LEJYONER", 0, 0, 0, 0, 500));
        }

        public void PrepareSatellites()
        {
            foreach (var satellite in equippedStationModule)
            {
                var center = Position;
                Position position = null;
                string name = "";

                switch (satellite.slotId)
                {
                    case 1:
                        position = new Position(center.X - 413, center.Y - 98);
                        name = "M-01";
                        break;
                    case 2:
                        position = new Position(center.X - 171, center.Y - 236);
                        name = "M-02";
                        break;
                    case 3:
                        position = new Position(center.X + 170, center.Y + 236);
                        name = "M-03";
                        break;
                    case 4:
                        position = new Position(center.X + 412, center.Y - 98);
                        name = "M-04";
                        break;
                    case 5:
                        position = new Position(center.X + 412, center.Y + 97);
                        name = "M-05";
                        break;
                    case 6:
                        position = new Position(center.X + 170, center.Y - 235);
                        name = "M-06";
                        break;
                    case 7:
                        position = new Position(center.X - 171, center.Y + 235);
                        name = "M-07";
                        break;
                    case 8:
                        position = new Position(center.X - 413, center.Y + 97);
                        name = "M-08";
                        break;
                }

                new Satellite(Spacemap, name, FactionId, position, Clan);
            }
        }

        public override short GetAssetType() { return Clan == null ? ASSET_TYPE_NO_CLAN : ASSET_TYPE_CLAN_OWNING; }

        public override void Click(GameSession gameSession)
        {
            var player = gameSession.Player;
            player.SendCommand(BattleStationBuildingUiInitializationCommand.write(Id, Id, "Test",
                                                                                      new AsteroidProgressCommand(
                                                                                              Id, 
                                                                                              (float)0.1,
                                                                                              (float)0.1, 
                                                                                              "En iyi", 
                                                                                              "Mevcut clan",
                                                                                              new EquippedModulesModule(equippedStationModule),
                                                                                              true),
                                                                                      new AvailableModulesCommand(inventoryStationModule),
                                                                                      1, 
                                                                                      1, 
                                                                                      1));
        }

        public override byte[] GetAssetCreateCommand()
        {
            return AssetCreateCommand.write(new AssetTypeModule(GetAssetType()), ASSET_NAME,
                                          FactionId, (Clan != null ? Clan.Tag : ""), Id, DESIGN_ID, 3,
                                          Position.X, Position.Y, (Clan != null ? Clan.Id : 0), true, true, true,
                                          new ClanRelationModule(ClanRelationModule.NONE),
                                          new List<VisualModifierCommand>());
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/map_entities/stationary/stations/BattleStation.java
package simulator.map_entities.stationary.stations;


import net.game_server.GameSession;

import java.util.Vector;
import java.util.concurrent.atomic.AtomicInteger;

import simulator.map_entities.stationary.ActivatableStationaryMapEntity;
import simulator.netty.ServerCommand;
import simulator.netty.serverCommands.AssetCreateCommand;
import simulator.netty.serverCommands.AssetTypeModule;
import simulator.netty.serverCommands.AsteroidProgressCommand;
import simulator.netty.serverCommands.AvailableModulesCommand;
import simulator.netty.serverCommands.BattleStationBuildingUiInitializationCommand;
import simulator.netty.serverCommands.ClanRelationModule;
import simulator.netty.serverCommands.EquippedModulesModule;
import simulator.netty.serverCommands.ShipWarpModule;
import simulator.netty.serverCommands.StationModuleModule;
import simulator.netty.serverCommands.VisualModifierCommand;
import simulator.system.SpaceMap;
import simulator.system.clans.Clan;
import utils.Log;

/**
 Created by Pedro on 09-04-2015.
 */
public class BattleStation
        extends ActivatableStationaryMapEntity {

    private static final short ASSET_TYPE_CLAN_OWNING = AssetTypeModule.BATTLESTATION;
    private static final short ASSET_TYPE_NO_CLAN     = AssetTypeModule.BATTLESTATION;
    private static final short DESIGN_ID              = 0;

    private final AtomicInteger itemIds = new AtomicInteger(0);

    private Clan mClan;

    public BattleStation(final SpaceMap pCurrentSpaceMapId, final int pPositionX, final int pPositionY) {
        super(pCurrentSpaceMapId);

        this.setCurrentPositionX(pPositionX);
        this.setCurrentPositionY(pPositionY);
    }

    @Override
    public void handleClick(final GameSession pGameSession) {
        Log.p("BattleStation Handle Click");
        if (this.getAssetType() == ASSET_TYPE_NO_CLAN) {
        	
        	Vector<StationModuleModule> modulee = new Vector<StationModuleModule>();
        	modulee.add(new StationModuleModule(1,5,5,StationModuleModule.LASER_LOW_RANGE,10000,10000,1000,1000,16,"TESTTT",0,0,0,0,0));
        	
            pGameSession.getGameServerClientConnection()
                        .sendToSendCommand(new BattleStationBuildingUiInitializationCommand(this.getMapEntityId(), 1, "Test",
                                                                                      new AsteroidProgressCommand(
                                                                                              this.getMapEntityId(), 1,
                                                                                              1, "test2", "test3",
                                                                                              new EquippedModulesModule(
                                                                                            		  modulee),
                                                                                              true),
                                                                                      this.getBattleStationAvailableModulesCommand(),
                                                                                      0, 0, 0));
            Log.p("BatteStationBuildingUi Command Sent");
        }
    }

    private AvailableModulesCommand getBattleStationAvailableModulesCommand() {
        final Vector<StationModuleModule> stationModuleModules = new Vector<>();
        stationModuleModules.add(new StationModuleModule(1,5,5,StationModuleModule.LASER_LOW_RANGE,10000,10000,1000,1000,16,"TESTTT",0,0,0,0,0));
        return new AvailableModulesCommand(stationModuleModules);
    }

    @Override
    public String getAssetName() {
        return "Clan Battle Station";
    }

    @Override
    public short getAssetType() {
        return mClan == null ? ASSET_TYPE_NO_CLAN : ASSET_TYPE_CLAN_OWNING;
    }

    @Override
    public ServerCommand getAssetCreateCommand() {
        Log.p(this.getMapEntityId() + "");
        return new AssetCreateCommand(new AssetTypeModule(this.getAssetType()), this.getAssetName(),
                                      this.getFactionId(), "", this.getMapEntityId(), DESIGN_ID, 9990,
                                      this.getCurrentPositionX(), this.getCurrentPositionY(), 9991, true, true, true, true,
                                      new ClanRelationModule(ClanRelationModule.NONE),
                                      new Vector<VisualModifierCommand>());
    }

    private short getFactionId() {
        if (this.mClan != null) {
            return this.mClan.getFactionId();
        }
        return 0;
    }

    public Clan getClan() {
        return this.mClan;
    }

    public short getStatus() {
        return 0;
    }
}
